### PR TITLE
python312Packages.uamqp: 1.6.9 -> 1.6.11

### PR DESCRIPTION
**Pull request by natsukium**
Diff: https://github.com/Azure/azure-uamqp-python/compare/refs/tags/v1.6.9...v1.6.11

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 353152` at 0d0d64b1 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.taxi</li>
    <li>taxi-cli (python312Packages.taxi)</li>
  </ul>
</details>

**Comment by tjni** [comment]
Can you try pulling in https://github.com/Azure/azure-uamqp-python/pull/411 and rebuilding? Currently, it fails for me on aarch64-darwin due to https://github.com/Azure/azure-uamqp-python/issues/386.

File: pkgs/development/python-modules/uamqp/default.nix [hunk 1]
```diff
@@ -17,41 +17,19 @@
...
-        # builder anyway
-        includes = [ "src/vendor/azure-uamqp-c/src/link.c" ];
-        hash = "sha256-EqDfG1xAz5CG8MssSSrz8Yrje5qwF8ri1Kdw+UUu5ms=";
-      })
-      # Fix incompatible function pointer conversion error with clang 16.
-      ./clang-fix-incompatible-function-pointer-conversion.patch
-    ];
+  patches = lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
+    ./darwin-azure-c-shared-utility-corefoundation.patch
+  ];
```
Comment (tjni) [inline pkgs/development/python-modules/uamqp/default.nix:32]: I don't have x86_64-darwin so can't test, but I wonder if this is still needed even after the Darwin rework. (It sucks that we have to wait for 9-10 hours for an ofborg eval.)

Comment (natsukium) [reply pkgs/development/python-modules/uamqp/default.nix]: I removed the patch as it didn't seem to be needed anymore. Thanks.

**Comment by natsukium** [comment]
> Can you try pulling in https://github.com/Azure/azure-uamqp-python/pull/411 and rebuilding? Currently, it fails for me on aarch64-darwin due to https://github.com/Azure/azure-uamqp-python/issues/386.

I accidentally deleted the patch that was there. Thanks for submitting the PR.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 353152` at b8b2bef7 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.uamqp</li>
    <li>python312Packages.uamqp</li>
  </ul>
</details>

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 353152` at b8b2bef7 run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.uamqp</li>
    <li>python312Packages.uamqp</li>
  </ul>
</details>

**Review by tjni — APPROVED** [review]

**Pull request merged**